Style/custom dark theme

stvmachine opened the pull request:
## Preview
[https://stvmachine.github.io/osv.dev/quickstart/](https://stvmachine.github.io/osv.dev/quickstart/)

## Description

Using a dark theme for the docs as a starting point. Just-the-docs come with a [default dark theme](https://stvmachine.github.io/osv.dev/quickstart/).

The updates:
- **General styles have been updated.** More information on the Figma updates: https://www.figma.com/design/Yi1nd5v6iR406zngmscSD3/Open-Source-Vulnerability-(OSV)-Scanner?node-id=0-1&t=ca4V36K9Cf8KrG4O-0

- **The logo has been updated**

![Screenshot 2024-07-24 at 12 50 12 PM](https://github.com/user-attachments/assets/38dd6051-dbfa-4d8f-9d6a-819408e59b8d)

**Figure 1**: Updated logo.

- **Code snippet highlight has been updated**: Using [native](https://github.com/dwayne/sass-pygments/blob/master/styles/scss/native.scss) due to the default dark theme having issues with comments on JSON due they are recognized and styled as errors. The code snippets can be changed to any other pygment theme as long we have the scss file.

## Screenshots
![Screenshot 2024-07-26 at 12 20 03 PM](https://github.com/user-attachments/assets/039fb593-b67e-4a14-9ed5-b6faeba8dd12)
![Screenshot 2024-07-26 at 11 55 08 AM](https://github.com/user-attachments/assets/459cb8d2-1891-482e-a39d-a3e0c76a67de)
![Screenshot 2024-07-25 at 3 03 36 PM](https://github.com/user-attachments/assets/02abec70-d791-43cf-96e6-08aa51159a84)
![Screenshot 2024-07-26 at 11 54 42 AM](https://github.com/user-attachments/assets/03dc99cb-fa83-447d-8078-a142e945ef8e)
![Screenshot 2024-07-26 at 11 54 58 AM](https://github.com/user-attachments/assets/5275674a-22ff-46f4-9513-3637cdc89850)

another-rex commented:
The colours seem a bit off, specifically I think the background colour should match the background of OSV.dev / figma design.

stvmachine commented:
> The colours seem a bit off, specifically I think the background colour should match the background of OSV.dev / figma design.

Yeah, you are right. My mistake, I made some assumptions in terms of black/white. I double-checked some of the visible styles on Figma versus the updates and at least those should match now.

Styles for tables, search input & search results, code snippet styles are not covered by Figma and unfortunately, I just went with defaults plus this solarised dark theme for code highlight. Not sure, how to iterate over this work more than involve more @joshbuker and relay on his input 🙏🏼

another-rex commented:
I'm not a big fan of the solarized-dark code theme, on our blog pages we use the chroma 'native' theme: https://xyproto.github.io/splash/docs/native.html 

It looks like there's a scss version of it here: https://github.com/dwayne/sass-pygments/blob/master/styles/scss/native.scss

Can you try using that instead?

As for the search input, using the same background colour as the main page (#2A2A2A) should be fine, as the entire background darkens anyway when the search is selected.

stvmachine commented:
> I'm not a big fan of the solarized-dark code theme, on our blog pages we use the chroma 'native' theme: https://xyproto.github.io/splash/docs/native.html
> 
> It looks like there's a scss version of it here: https://github.com/dwayne/sass-pygments/blob/master/styles/scss/native.scss
> 
> Can you try using that instead?
> 
> As for the search input, using the same background colour as the main page (#2A2A2A) should be fine, as the entire background darkens anyway when the search is selected.

Updates:
- Search-input background-color changed to #2A2A2A
- Updated table background-color. [Preview](https://stvmachine.github.io/osv.dev/post-v1-query/). I came up with the lighter version after doing some math similar to the default gradient. Let me know if the darker version works for you.

✅  Lighter version (on use)
![Screenshot 2024-07-26 at 12 11 09 PM](https://github.com/user-attachments/assets/83c100a3-5a2f-4a68-9d63-44ce18b0cf23)

🔴  Dark version (not used)
![Screenshot 2024-07-26 at 12 11 01 PM](https://github.com/user-attachments/assets/ba4fac3a-8998-4b84-8529-ad3729f4241a)

- Replaced code snippet with native. Updated screenshot at top but can share here as well
![Screenshot 2024-07-26 at 11 54 58 AM](https://github.com/user-attachments/assets/bb1a433c-f627-4c19-b93c-e6d5d5485a65)

another-rex reviewed (approved):
Looks great! Happy with the light background version for tables

another-rex commented:
> Did you mean to ping me?

Ah sorry, I don't think so, we have someone with the same name internally and @ is getting confused.

another-rex commented:
@stvmachine I just realised that the background color of OSV.dev is actually #292929, can you see if you can update the 2A2A2A to 292929? Thanks!

another-rex commented:
osv.dev also uses the Overpass font, can you also try switching from the current roboto font to overpass and see how that looks?

another-rex commented:
Thanks, just need to rebase/merge this from master and we should be good to merge.

another-rex commented on gcp/appengine/docs/index.html:
```diff
@@ -5,7 +5,7 @@
     <meta charset="utf-8"/>
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link href="https://fonts.googleapis.com/css?family=Montserrat:300,400,700|Roboto:300,400,700" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Montserrat:300,400,700|Overpass:300,400,700" rel="stylesheet">
```
We also need to set this font in the CSS, not just adding the resource. I believe it defaulted to roboto because that's the default system font.

oliverchang replied:
Friendly ping on this @stvmachine @G-Rath

another-rex reviewed (commented):
I think the font is still incorrect, we need to specify the font used in the CSS as well. Also please merge in the master branch to get the PR check to pass.
